### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by sangwa7**
- [ ]  Installed and setup devise
- [ ]  Implemented registering a new user.
- [ ]  Login in a user with a combination of email and password.
- [ ]  Hashed password and stored it in the database.
- [ ]  Implemented ask for confirmation of email.
- [ ]  Implemented reset password.
- [ ]  Modified the views in devise
- [ ]  Added styling

And below we have included the screenshots of our project working and the login page with the added data to the database.

image

image

image

image

image

image

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,56 +1,36 @@
...
 
   def new
-    @post = Post.new
-
+    @new_post = Post.new
   end
 
   def create
-    post = params[:post]
+    puts params
```
Comment (Deepakdanger) [inline app/controllers/posts_controller.rb]: - [ ] Kindly remove this line.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,56 +1,36 @@
...
+    post.comments_counter = 0
+    post.likes_counter = 0
+    if post.save
+      flash[:notice] = 'Your post has been saved'
+      redirect_to user_posts_url
     else
-      flash[:error] = 'Creating new post failed!'
-      @post = new_post
-      render :new
+      flash[:alert] = 'Your comment has been saved'
```
Comment (Deepakdanger) [inline app/controllers/posts_controller.rb]: - [ ] Kindly post the meaningful alert.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Team, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 
- Good uses of devise. :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (rloterh) [reply app/controllers/posts_controller.rb]: This issue is well resolved! ✔️

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (rloterh) [reply app/controllers/posts_controller.rb]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Didier,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉🌟 

## Highlights 👍
- Authentication using devise gem properly implemented


<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**